rust/diesel/src/order.rs

```diff
@@ -109,16 +115,29 @@ impl Order {
                     .execute(connection)
             });
 
-        update_result.map(|_| order).ok()
+        match update_result {
+            Err(diesel::result::Error::NotFound) | Ok(0) => Ok(None),
```

Comment:
I'm not super familiar with diesel, but it seems strange that there are two different cases that both mean the same thing here. What does the `0` represent here?

Reply:
`Ok(x)` means that SQL was executed successfully, where `x` is number of affected rows.
Eg. if an entity selected by SQL does not exist, it will return Ok(0)

Reply:
Ahh I see, that makes sense.